framework/include/utils/MooseRandom.h

```diff
@@ -153,19 +151,23 @@ class MooseRandom
    */
   void saveState()
   {
-    for (LIBMESH_BEST_UNORDERED_MAP<unsigned int, std::pair<mt_state, mt_state> >::iterator it = _states.begin();
-         it != _states.end(); ++it)
-      it->second.second = it->second.first;
+    std::for_each(_states.begin(), _states.end(),
+                  [](std::pair<const unsigned int, std::pair<mt_state, mt_state>> & pair)
+                    {
+                      pair.second.second = pair.second.first;
+                    });
```

Comment:
I don't know if I'm crazy about our lambda indentation style... but I'm not sure there even *is* a good lambda indentation style.